sopel/modules/remind.py

```diff
@@ -226,47 +215,219 @@ def remind_in(bot, trigger):
+        :type today: :class:`datetime.datetime`
+        :return: The duration, in second, between ``today`` and the reminder.
+        :rtype: :class:`int`
+
+        This method returns the number of seconds given by the
+        :class:`datetime.timedelta` obtained by the difference between the
+        reminder and ``today``.
+
+        If the delta between the reminder and ``today`` is negative, Python
+        will represent it as a negative number of days, and a positive number
+        of seconds: since it returns the number of seconds, any past reminder
+        will be transformed into a future reminder the next day.
+        """
+        if not today:
+            today = datetime.now(self.timezone)
+        else:
+            today = today.astimezone(self.timezone)
+
+        year = today.year
+        month = today.month
+        day = today.day
+
+        if self.date1 and self.date2 and self.date3:
+            if len(self.date1) == 4:
+                # YYYY-mm-dd
+                year = int(self.date1)
+                month = int(self.date2)
+                day = int(self.date3)
+            else:
+                # dd-mm-YYYY or dd/mm/YY
+                year = int(self.date3)
+                month = int(self.date2)
+                day = int(self.date1)
+        elif self.date1 and self.date2:
+            if len(self.date1) == 4:
+                # YYYY-mm
+                year = int(self.date1)
+                month = int(self.date2)
+            elif len(self.date2) == 4:
+                # mm-YYYY
+                year = int(self.date2)
+                month = int(self.date1)
+            else:
+                # dd/mm
+                month = int(self.date2)
+                day = int(self.date1)
+
+        at_time = datetime(
+            year, month, day,
+            self.hour, self.minute, self.second,
+            tzinfo=today.tzinfo)
+
+        timediff = at_time - today
+        duration = timediff.seconds
+
+        if timediff.days > 0:
+            duration = duration + (86400 * timediff.days)
+
+        return duration
+
+
+def parse_regex_match(match, default_timezone=None):
+    """Parse a time reminder from ``match``
+
+    :param match: :obj:`~.REGEX_AT`'s matching result
+    :param default_timezone: Default trigger's timezone
+                             (from the nick's, channel's, or config's timezone)
+    :rtype: :class:`TimeReminder`
+    """
+    try:
+        timezone = validate_timezone(match.group('tz') or 'UTC')
+    except ValueError:
+        timezone = default_timezone or 'UTC'
+
+    return TimeReminder(
+        int(match.group('hours')),
+        int(match.group('minutes')),
+        int(match.group('seconds') or '0'),
+        timezone,
+        match.group('date1'),
+        match.group('date2'),
+        match.group('date3'),
+        match.group('message')
+    )
 
 
 @module.commands('at')
 @module.example('.at 13:47 Do your homework!')
```

Comment:
This deserves more examples. With #1403 merged, more than one example can appear in the help output.

Reply:
Good point. I remembered that there was a PR in progress, but totally forgot to check if it was merged. I was way too focus on unit-testing that.